### PR TITLE
Do not prompt for close on import

### LINKED ISSUE #1460: Importing a file: Are you sure you want to cancel
Clicking on Import does the following

image

WHAT?

### PR DESCRIPTION
**Pull request by abdelr**
Fixes #1460 (Do not prompt for close on import)

### CODE REVIEW

**Comment by msevestre** [comment]
What does the view.CloseOnImport do? Why is the presenter asking the view to trigger the close? This feels weird.

Also needs test

**Review by msevestre — CHANGES_REQUESTED** [review]
Test missing and wondering about flow between view and presenter. Going to check the code

File: src/OSPSuite.Presentation/Presenters/Importer/IModalImporterPresenter.cs [hunk 1]
```diff
@@ -41,10 +41,13 @@ public ModalImporterPresenter(IModalImporterView view, IImporterPresenter import
          AddSubPresenters(importerPresenter);
       }
 
+      private bool _shouldPromtBeforeClose = false;
```
Comment (msevestre) [inline src/OSPSuite.Presentation/Presenters/Importer/IModalImporterPresenter.cs:44]: move the definition of this flag at the top (use resharper) and correct typo (Prompt)

File: src/OSPSuite.Presentation/Presenters/Importer/IModalImporterPresenter.cs [hunk 2]
```diff
@@ -106,7 +109,12 @@ ImporterConfiguration configuration
             results.Each(r => r.ConfigurationId = configurationId);
          }
 
-         _importerPresenter.OnTriggerImport += (s, d) => { _view.CloseOnImport(); };
+         _importerPresenter.OnTriggerImport += (s, d) => 
+         {
+            _shouldPromtBeforeClose = false;
+            _view.CloseOnImport();
```
Comment (msevestre) [inline src/OSPSuite.Presentation/Presenters/Importer/IModalImporterPresenter.cs:115]: Oh see. This methods already exists and it is called CloseView. 
From the name it looked it was triggering some event.. (Onxxx) .Naming is really everything in programming :)

Comment (IndrajeetPatil) [reply src/OSPSuite.Presentation/Presenters/Importer/IModalImporterPresenter.cs:115]: > "There are only two hard things in Computer Science: cache invalidation and naming things"?
> Phil Karlton

😅

**Comment by abdelr** [comment]
@msevestre the test is not valid because the view.Display() method returns immediately. So, this is a blocking scenario. I have some ideas on how to implement this but I wonder if we already have some support for it. We could talk later about it if you want.

**Comment by msevestre** [comment]
A few remarks:
First,why are you registering the event outside of the constructor?
Second, why are you registering to the event twice?
Last we never have this type of boolean hanging around to decide if we should close or not. Any idea why this is required here?

**Review by msevestre — COMMENTED** [review]
Is the test ok now? @abdelr

**Comment by abdelr** [comment]
We still do not have a cancel button, but I am not sure if we want to change this. In any case, if we do, it will be just adding the button which will invoke form close but keep in mind that we already have some buttons changing captions and could be a bit confusing. We have the X button anyways.

**Pull request merged**